### PR TITLE
Fix Access Denied error on accessing notifications/<id> and email-notifications/<id>

### LINKED ISSUE #4127: endpoint /notification/<id> and email-notification/<id> gives Access Denied error
**I'm submitting a ...**  (check one with "x")
- [X] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

**Current behavior:**

**Expected behavior:**

**Steps to reproduce:**

GET `notifications/<id>` and GET `email-notifications/<id>`

**Other information:**

Due to permission checking `user.id = kwargs['id']` for all `is_user_itself` decorator.
**Note:** In this case `id` is notification's id rather than user's id, so check condition is not valid

### PR DESCRIPTION
**Pull request by magdalenesuo**
Fixes #4127

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Fixed decorators for notification and email notification api

### CODE REVIEW

**Comment by magdalenesuo** [comment]
Travis should pass after #4080 is merged

File: app/api/email_notifications.py [hunk 1]
```diff
@@ -91,7 +93,8 @@ class EmailNotificationDetail(ResourceDetail):
     """
     Email notification detail by ID
     """
-    decorators = (is_user_itself,)
+    decorators = (api.has_permission('is_user_itself', fetch="user_id", fetch_as="id",
+                  model=EmailNotification, fetch_key_url="id"),)
```
Comment (poush) [inline app/api/email_notifications.py]: Not an issue but fetch_key_url is not required parameter and by default it is "id"

File: app/api/notifications.py [hunk 2]
```diff
@@ -87,7 +88,8 @@ class NotificationDetail(ResourceDetail):
     """
     Notification detail by ID
     """
-    decorators = (is_user_itself,)
+    decorators = (api.has_permission('is_user_itself', fetch="user_id", fetch_as="id",
+                  model=Notification, fetch_key_url="id"),)
```
Comment (poush) [inline app/api/notifications.py]: One here also :P

File: app/factories/email_notification.py [hunk 3]
```diff
@@ -14,3 +14,5 @@ class Meta:
     session_accept_reject = 0
     session_schedule = 1
     after_ticket_purchase = 1
+    event_id = 1
```
Comment (poush) [inline app/factories/email_notification.py]: You forgot to create the event relationship here

**Pull request merged**